client/components/happychat/composer.jsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+/**
+ * Internal dependencies
+ */
+import {
+	updateChatMessage,
+	sendChatMessage
+} from 'state/happychat/actions';
+import {
+	when,
+	forEach,
+	compose,
+	propEquals,
+	call,
+	prop
+} from './functional';
+import scrollbleed from './scrollbleed';
+import { translate } from 'i18n-calypso';
+
+// helper function for detecting when a DOM event keycode is pressed
+const returnPressed = propEquals( 'which', 13 );
+// helper function that calls prevents default on the DOM event
+const preventDefault = call( 'preventDefault' );
+
+/*
+ * Renders a textarea to be used to comopose a message for the chat.
+ */
+export const Composer = React.createClass( {
+	mixins: [ scrollbleed ],
```

Comment:
In a future PR, we'll probably want to move away from mixins here since they're discouraged and not compatible with es6 classes (https://facebook.github.io/react/docs/reusable-components.html#no-mixins)